File: Jessica/107.Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
# 自顶向下递归,每次加入深度信息
class Solution:
    def Recursion(self, node:TreeNode, outter:List[List[int]], depth:int):
        if len(outter)<depth:
            outter.append([])
        outter[depth-1].append(node.val)
        if node.left:
            self.Recursion(node.left,outter,depth+1)
        if node.right:
            self.Recursion(node.right,outter,depth+1)   
        
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        if root == None:
            return []
        outter = [[]]
        self.Recursion(root,outter,1)
        #return list(reversed(outter))
        return outter[::-1]

File: Jessica/107.Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
       void search(TreeNode* node, vector<vector<int>>& vec, int depth) {
       
        if (vec.size() < depth) vec.push_back({});
        
       
        vec[depth - 1].push_back(node->val);
        
        if(node->left) search(node->left, vec, depth + 1);
        if(node->right) search(node->right, vec, depth + 1);
    }
    
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        
        vector<vector<int>> ans;
        if (root)search(root, ans, 1);
       
        
        std::reverse(ans.begin(), ans.end());
        return ans;
    }
};
